slashCommands/vote.ts

```diff
@@ -12,187 +21,149 @@ export default {
+            }
+            if (!timed && pro.size + con.size == usedVotes) {
+                reply.edit(getEmbedOptions(title, msg, Math.ceil(Date.now()/1000 + 30)))
+                setTimeout(() => {
+                    printVotes(pro, con, reply, title, collector);
+                }, 30000 );
             }
-            reactionAr = tempAr;
         });
 
-        //delay loop
-        for (let i = 0; i < maxLoop * time; i++) {
-            await delay(1000);
-            if (reactionAr.length >= usedVotes && timed == false) {
-                voteAN.addFields({ name: "Enough Votes", value: "Vote ending in " + time + " Seconds" });
-                await reply.edit({embeds: [voteAN]});
-                await delay(time * 1000);
-                break;
-            }
+        if (timed) {
+            setTimeout(() => {
+                printVotes(pro, con, reply, title, collector);
+            }, 1000*time);
         }
 
-        //stop the collector
-        collector.stop();
-
-
-        //counting up and down votes and creating the text
-        await printVotes(reactionAr, reply, title)
     }
 } as ISlashCommand;
 
-//delays a function
-function delay(ms: number) {
-    return new Promise(resolve => setTimeout(resolve, ms));
-}
-
-async function printVotes(reactionAr, reply, title) {
-    let upVotes = "";
-    let downVotes = "";
-
-    let up = 0;
-    let down = 0;
 
-    for (const i of reactionAr) {
-        if (i[0] == -1) {
-            downVotes += ("🔴 " + i[1] + "\n");
-            down += 1;
-        } else {
-            upVotes += ("🟢 " + i[1] + "\n");
-            up += 1;
-        }
-    }
+async function printVotes(pro: Set<string>, con: Set<string>, reply: Message, title: string, collector: InteractionCollector<any>): Promise<void> {
+    collector.stop();
 
+    let upVotes = Array.from(pro).map(e => {
+        return "🟢 <@" + e + ">";
+    }).join("\n");
+    let downVotes = Array.from(con).map(e => {
+        return "🔴 <@" + e + ">";
+    }).join("\n");
+    
     upVotes = (upVotes == "") ? "None" : upVotes;
     downVotes = (downVotes == "") ? "None" : downVotes;
 
-    //remove all reactions
-
-    await reply.reactions.removeAll();
-
     //finished embed
-
     const msgEmbed = new EmbedBuilder()
-        .setColor((down < up) ? "#02B22E" : "#cc0000")
-        .setTitle((down < up) ? "Vote accepted" : "Vote failed")
+        .setColor((con.size < pro.size) ? "#02B22E" : "#cc0000")
+        .setTitle((con.size < pro.size) ? "Vote accepted" : "Vote failed")
         .addFields({name: "For", value: upVotes, inline: true}, {name: "Against", value: downVotes, inline: false})
         .setTimestamp();
 
     //if a title were specified it will be displayed
-
     if (title != "") {
         msgEmbed.setDescription(title);
     }
-
     //show embed
     await reply.edit({embeds:[msgEmbed]})
+}
+
+function getEmbedOptions(title:string, msg: string, timestamp?:number):{embeds, components, fetchReply}{
```

Comment:
This can just return the type `InteractionReplyOptions`

Reply:
'InteractionReplyOptions' is not assignable to parameter of type 'string | MessagePayload | MessageEditOptions' ; it also seems like MessagePayload dont support Embeds and MessageEditOptions dont support fetchReply.